protopipe/mva/utils.py

```diff
@@ -18,21 +18,45 @@ def load_obj(name):
         return pickle.load(f)
 
 
-def prepare_data(ds, cuts, label=None):
-    """Add variables in data frame"""
-    ds["log10_hillas_intensity"] = np.log10(
-        ds["hillas_intensity_reco"]
-    )  # THIS SHOULDN'T BE HARDCODED!!!
-    ds["log10_impact_dist"] = np.log10(ds["impact_dist"])
+def prepare_data(ds, derived_features, cuts, select_data=True, label=None):
+    """Add custom variables to the input data and optionally select it.
+
+    Parameters
+    ----------
+    ds : pandas.DataFrame
+        Input data not yet selected.
+    derived_features: dict
+        Dictionary of more complex featuresread from the configuration file.
+    cuts: str
+        Fiducial cuts from protopipe.mva.utils.make_cut_list
+    select_data: bool
+        If True apply cuts to the final dataframe.
+    label: str
+        Name of the classifier target label if any.
+
+    Returns
+    -------
+    ds : pandas.DataFrame
+        Input data integrated with new variables and optionally selected for
+        the fiducial cuts.
+    """
+
     ds["log10_true_energy"] = np.log10(ds["true_energy"])
-    try:  # for classification
+
+    try:  # additional parameters for classification
         ds["log10_reco_energy"] = np.log10(ds["reco_energy"])
         ds["log10_reco_energy_tel"] = np.log10(ds["reco_energy_tel"])
         ds["label"] = np.full(len(ds), label)
-    except:
-        pass
+    except KeyError as e:
+        print(e)
```

Comment:
should this really silently fail? If so, maybe add a comment explaining why the error is caught but nothing is done with it.

Reply:
Looking at it a second time I decided that it was not done well so I rewrote it.

basically now:

- if label is None, it means that we are doing an energy regressor, so all this code block doesn't happen
- if it's not None, label is always added to the dataframe, but the 2 energy keys are checked for existence since in our reference analysis we need energy as 1 of the features

unfortunately right now this is needed to make DL2 work here, so I raise an error if the model is a classifier and those keys are not selected for usage (or I could add them always?)